This is the diff I apply to the md5c.c found in the python tarball
to get my local version that works for iTunes.

--- md5c.c.orig	2007-03-12 14:00:10.000000000 +0000
+++ md5c.c	2007-02-27 16:57:47.000000000 +0000
@@ -45,6 +45,9 @@
 #define S43 15
 #define S44 21
 
+/* this defines us to be at modification level 1 */
+#define AppleModLevel   1
+
 static void MD5Transform(UINT4[4], unsigned char[64]);
 static void Encode(unsigned char *, UINT4 *, unsigned int);
 static void Decode(UINT4 *, unsigned char *, unsigned int);
@@ -206,7 +209,24 @@
     GG(a, b, c, d, x[ 9], S21, 0x21e1cde6); /* 25 */
     GG(d, a, b, c, x[14], S22, 0xc33707d6); /* 26 */
     GG(c, d, a, b, x[ 3], S23, 0xf4d50d87); /* 27 */
-    GG(b, c, d, a, x[ 8], S24, 0x455a14ed); /* 28 */
+
+    /* TOM - this big shamelessly cribbed from
+    http://search-dev.develooper.com/src/BILLH/Digest-MD5-M4p-0.01/M4p.xs */
+
+    /* begin apple md5 modification type 1 */
+    /* Apple does this to make it hard for us ordinary mortals */
+    if(AppleModLevel == 1) 
+    {
+        GG (b, c, d, a, x[ 8], S24, 0x445a14ed); /* 28 */
+    }
+    else /* real MD5 */
+    {
+        GG (b, c, d, a, x[ 8], S24, 0x455a14ed); /* 28 */
+    }
+    /* end apple md5 modification type 1, hopefully there will be no part 2 etc */
+    
+
+
     GG(a, b, c, d, x[13], S21, 0xa9e3e905); /* 29 */
     GG(d, a, b, c, x[ 2], S22, 0xfcefa3f8); /* 30 */
     GG(c, d, a, b, x[ 7], S23, 0x676f02d9); /* 31 */
